### PR TITLE
fix(lsp): don't try to process semantic tokens if the response is empty

### PR DESCRIPTION
**Pull request by hashinclude**
#21387

### CODE REVIEW

File: runtime/lua/vim/lsp/semantic_tokens.lua [hunk 1]
```diff
@@ -331,6 +331,10 @@ function STHighlighter:process_response(response, client, version)
     tokens = response.data
```
Comment (folke) [inline runtime/lua/vim/lsp/semantic_tokens.lua:331]: Probably be better to change this line instead to `tokens = response.data or {}`?

**Review by folke — COMMENTED** [review]
What lsp server gives you these empty tokens result?

Reading the LSP spec, this should not happen. Either a `edits` should be present, or a `data`. That's also how the current code handles the response.

See https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens

**Comment by hashinclude** [comment]
> What lsp server gives you these empty tokens result?

It's an internal LSP server, which is probably misbehaving occasionally.

**Comment by folke** [comment]
@jdrouhard wdyt? Shouldn't happen normally, since according to the spec either `edits` or `data` should be present.

**Comment by mfussenegger** [comment]
The server needs to be fixed,  we don't add workarounds for clear spec violations.

What we could do is add an assert to have an error message that tells the user that the server is broken

**Comment by jdrouhard** [comment]
Yeah, I agree with @mfussenegger - no workarounds for spec violations but we could try for better error messages when we do detect something invalid.

**Comment by justinmk** [comment]
Could add an `assert()` for this specific case. But more likely we need a general validator, perhaps driven by the LSP [metaModel](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#metaModel)

**Pull request closed without merging**